test/integration/02-user-wallet/02-b-stablesats.spec.ts

```diff
@@ -0,0 +1,323 @@
+
+  const convertFromBtc = priceRatio.convertFromBtc(btc)
+  const convertFromBtcToFloor = priceRatio.convertFromBtcToFloor(btc)
+  const convertFromBtcToCeil = priceRatio.convertFromBtcToCeil(btc)
+
+  return {
+    btc,
+    convertFromBtc,
+    convertFromBtcToFloor,
+    convertFromBtcToCeil,
+
+    getCentsFromSatsForImmediateBuy,
+    getCentsFromSatsForImmediateSell,
+    getCentsFromSatsForFutureBuy,
+    getCentsFromSatsForFutureSell,
+  }
+}
+
+const satsFromCents = async (usd: UsdPaymentAmount) => {
+  // Conversions with spreads
+  const getSatsFromCentsForImmediateBuy =
+    await newDealerFns.getSatsFromCentsForImmediateBuy(usd)
+  if (getSatsFromCentsForImmediateBuy instanceof Error) {
+    throw getSatsFromCentsForImmediateBuy
+  }
+
+  const getSatsFromCentsForImmediateSell =
+    await newDealerFns.getSatsFromCentsForImmediateSell(usd)
+  if (getSatsFromCentsForImmediateSell instanceof Error) {
+    throw getSatsFromCentsForImmediateSell
+  }
+
+  const getSatsFromCentsForFutureBuy = await newDealerFns.getSatsFromCentsForFutureBuy(
+    usd,
+  )
+  if (getSatsFromCentsForFutureBuy instanceof Error) throw getSatsFromCentsForFutureBuy
+
+  const getSatsFromCentsForFutureSell = await newDealerFns.getSatsFromCentsForFutureSell(
+    usd,
+  )
+  if (getSatsFromCentsForFutureSell instanceof Error) throw getSatsFromCentsForFutureSell
+
+  // Mid-rate conversions
+  const priceRatio = await newDealerFns.getCentsPerSatsExchangeMidRate()
+  if (priceRatio instanceof Error) throw priceRatio
+
+  const convertFromUsd = priceRatio.convertFromUsd(usd)
+
+  return {
+    usd,
+    convertFromUsd,
+
+    getSatsFromCentsForImmediateBuy,
+    getSatsFromCentsForImmediateSell,
+    getSatsFromCentsForFutureBuy,
+    getSatsFromCentsForFutureSell,
+  }
+}
+
+describe("define stablesats expected behavior", () => {
+  it("is the expected nominal rate", async () => {
+    const priceRatio = await newDealerFns.getCentsPerSatsExchangeMidRate()
+    if (priceRatio instanceof Error) throw priceRatio
+
+    expect(priceRatio.usdPerSat()).toEqual(0.02)
+  })
+
+  it("large values, round exact", async () => {
+    // ==========
+    // BTC TO USD
+    // ==========
+    const BTC = { amount: 10_000_000n, currency: WalletCurrency.Btc } as BtcPaymentAmount
+    const centsFromSatsRes = await centsFromSats(BTC)
+
+    // Mid-rate values
+    expect(centsFromSatsRes.convertFromBtc.amount).toEqual(200_000n)
+    expect(centsFromSatsRes.convertFromBtcToFloor.amount).toEqual(200_000n)
+    expect(centsFromSatsRes.convertFromBtcToCeil.amount).toEqual(200_000n)
+
+    // Values with spreads
+    expect(centsFromSatsRes.getCentsFromSatsForImmediateBuy.amount).toEqual(199_800n)
+    expect(centsFromSatsRes.getCentsFromSatsForImmediateSell.amount).toEqual(200_200n)
+    expect(centsFromSatsRes.getCentsFromSatsForFutureBuy.amount).toEqual(199_760n)
+    expect(centsFromSatsRes.getCentsFromSatsForFutureSell.amount).toEqual(200_200n)
```

Comment:
Should both `Sell` values be the same here?

Reply:
Should we merge these changes and then circle back to the stablesats potential issue, or should we wait for that upstream to be fixed?

Reply:
I guess we can merge - @enigbe can perhaps try to reproduce this in stablesats.